### ISSUE TITLE
App Platform image credentials set as required

### ISSUE THREAD

**Issue by davidsbond** [digitalocean/terraform-provider-digitalocean#1146]
# Bug Report

---

## Describe the bug
With the inclusion of https://github.com/digitalocean/terraform-provider-digitalocean/pull/1137 in the latest release the provider now throws an error if the `registry_credentials` string is not set. Previously, if you're using images within the DO registry, you don't need to provide credentials.

I don't think the `registry_credentials` field in the spec should be required.

### Affected Resource(s)

- digitalocean_app 

### Expected Behavior
Not including the `registry_credentials` property should not error when using a DO registry image or public image.

### Actual Behavior
You get the error:

```
 The argument "registry_credentials" is required, but no definition was found.
```

### Steps to Reproduce
Attempt to deploy a digitalocean app via terraform using a public image or something in the DO registry.

**Terraform Configuration Files**

**Terraform version**

**Debug Output**

**Panic Output**

## Additional context

**Important Factoids**

**References**

**Comment by pnduati** [comment]
cc @danaelhe

**Comment by danaelhe** [comment]
Hi there,

Thank you so much for bringing this to our attention. You are correct, that field should be optional. I will create a PR and cut a new release now.

**Comment by danaelhe** [comment]
Resolved in #1148 and just cut a new release with the fix in v2.37.1
